File: app/src/main/java/com/bugsnag/android/internal/StateObserver.java
package com.bugsnag.android.internal;

import com.bugsnag.android.StateEvent;

import androidx.annotation.NonNull;

public interface StateObserver {

    /**
     * This is called whenever the notifier's state is altered, so that observers can react
     * appropriately. This is intended for internal use only.
     */
    void onStateChange(@NonNull StateEvent event);
}

File: app/src/main/java/com/bugsnag/android/internal/StateObserver.kt
package com.bugsnag.android.internal

import com.bugsnag.android.StateEvent

fun interface StateObserver {
    /**
     * This is called whenever the notifier's state is altered, so that observers can react
     * appropriately. This is intended for internal use only.
     */
    fun onStateChange(event: StateEvent)
}
